### ISSUE TITLE
Will Android be supported in the future?

### ISSUE THREAD

**Issue by Codexiaoyi** [changkun/midgard#11]
Looking forward to Android.

**Comment by changkun** [comment]
Yes, Android can be easily supported. But I don’t have a good idea about how Android support should look like. For example, on desktop, there is a daemon process for syncing clipboard, on iOS, there are Shortcuts to fetch clipboard. But on Android, I can’t imagine a way to fetch them except developing an app. This seems less convenient.

If you have a good idea, feel free to suggest here.

**Comment by r4nchy** [comment]
@changkun You may want to look into [Tasker ](https://tasker.joaoapps.com/) android app, it is a trial/paid app which is kind of like iOS Shortcuts but on steroids. In it there is a clipboard change event trigger, and then there is a http request task that can be done.
I think it can be done on it.

**Comment by r4nchy** [comment]
it works on android 

**Midgard POST**
https://taskernet.com/shares/?user=AS35m8n2e1ZqSMAXi0rAdytI3lFROIIV6atUo8BPrMJuT45bsVYevMyh1FKJLYTk54TKvCKr&id=Profile%3AMidgard+SEND

**Midgard GET**
https://taskernet.com/shares/?user=AS35m8n2e1ZqSMAXi0rAdytI3lFROIIV6atUo8BPrMJuT45bsVYevMyh1FKJLYTk54TKvCKr&id=Profile%3AMidgard+RECEIVE

**Comment by changkun** [comment]
Thanks for the recommendation @r4nchy. It turns out that there is no system-level solution to support iOS-like "Shortcuts". Asking a user of Midgard to install a 3rd party app is not a goal can be aligned with user easily.

I think a feasible solution is to develop a midgard android app, using existing GUI solutions, such as Fyne, Gio, or GoMobile to support these features:

1. sync text clipboard content midgard server on android
2. allow upload pictures from android and upload to midgard server as image clipboard content
